mp4parse/src/lib.rs

```diff
@@ -1128,6 +1151,43 @@ fn read_stsc<T: Read>(src: &mut BMFFBox<T>) -> Result<SampleToChunkBox> {
     })
 }
 
+fn read_ctts<T: Read>(src: &mut BMFFBox<T>) -> Result<CompositionOffsetBox> {
+    let (version, _) = read_fullbox_extra(src)?;
+
+    let counts = be_u32(src)?;
+    let mut offsets = Vec::new();
+    for _ in 0..counts {
+        if src.bytes_left() == 0 {
+            return Err(Error::InvalidData("invalid data in 'ctts' box"));
+        }
```

Comment:
I meant that you know the size of the TimeOffset table entries, so you can calculate how many bytes will be read before entering the loop. That way you can return early.